sklearn/model_selection/tests/test_split.py

```diff
@@ -1883,15 +1884,171 @@ def test_random_state_shuffle_false(Klass):
+    # Should fail if there are more folds than groups
+    groups = np.array([1, 1, 1, 2, 2])
+    X = y = np.ones(len(groups))
+    with pytest.raises(
+        ValueError,
+        match="Cannot have number of folds=4 greater than the number of groups=2",
+    ):
+        next(GroupTimeSeriesSplit(n_splits=3).split(X, y, groups))
+
+
+def test_group_time_series_max_train_size():
+    groups = np.array(["a", "a", "a", "b", "b", "b", "c", "c", "c", "d", "d", "d"])
+    n_samples = len(groups)
+    X = y = np.ones(n_samples)
+    splits = GroupTimeSeriesSplit(n_splits=3, max_train_size=3).split(X, y, groups)
+    check_splits = GroupTimeSeriesSplit(n_splits=3, max_train_size=3).split(
+        X, y, groups
+    )
+    _check_time_series_max_train_size(splits, check_splits, max_train_size=3)
+
+    # Test for the case where the size of a fold is greater than max_train_size
+    check_splits = GroupTimeSeriesSplit(n_splits=3, max_train_size=2).split(
+        X, y, groups
+    )
+    _check_time_series_max_train_size(splits, check_splits, max_train_size=2)
+
+    # Test for the case where the size of each fold is less than max_train_size
+    check_splits = GroupTimeSeriesSplit(n_splits=3, max_train_size=5).split(
+        X, y, groups
+    )
+    _check_time_series_max_train_size(splits, check_splits, max_train_size=2)
+
+
+def test_group_time_series_non_overlap_group():
+    groups = np.array(
+        [
+            "a",
+            "a",
+            "a",
+            "a",
+            "a",
+            "a",
+            "b",
+            "b",
+            "b",
+            "b",
+            "b",
+            "c",
+            "c",
+            "c",
+            "c",
+            "d",
+            "d",
+            "d",
+        ]
+    )
+    gtss = GroupTimeSeriesSplit(n_splits=3)
+    splits = gtss.split(groups, groups=groups)
+    train, test = next(splits)
+    assert_array_equal(train, np.array([0, 1, 2, 3, 4, 5]))
+    assert_array_equal(test, np.array([6, 7, 8, 9, 10]))
+    assert_array_equal(groups[train], np.array(["a", "a", "a", "a", "a", "a"]))
+    assert_array_equal(groups[test], np.array(["b", "b", "b", "b", "b"]))
+
+    train, test = next(splits)
+    assert_array_equal(train, np.array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]))
+    assert_array_equal(test, np.array([11, 12, 13, 14]))
+    assert_array_equal(
+        groups[train], np.array(["a", "a", "a", "a", "a", "a", "b", "b", "b", "b", "b"])
+    )
+    assert_array_equal(groups[test], np.array(["c", "c", "c", "c"]))
+
+    train, test = next(splits)
+    assert_array_equal(
+        train, np.array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14])
+    )
+    assert_array_equal(test, np.array([15, 16, 17]))
+    assert_array_equal(
+        groups[train],
+        np.array(
+            ["a", "a", "a", "a", "a", "a", "b", "b", "b", "b", "b", "c", "c", "c", "c"]
+        ),
+    )
+    assert_array_equal(groups[test], ["d", "d", "d"])
+
```

Comment:
Could you please add at least one assertion that checks a case where there is more than one group present in the test set? I think that should involve a larger number of unique groups with a small number of samples per group (e.g. groups of 10 groups of 2 or 3 data points each for instance).

Reply:
Done . Added new test cases test_group_time_series_non_overlap_group_**(1 , 2, 3 )**